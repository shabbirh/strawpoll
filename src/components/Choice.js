import React from 'react';
import TextField from 'material-ui/TextField';
import Divider from 'material-ui/Divider';
import IconButton from 'material-ui/IconButton';
import ContentClear from 'material-ui/svg-icons/content/clear';

const styles = {
  container: {position:'relative', paddingRight:'48px'},
  icon: {position:'absolute', right:0},
};

class Choice extends React.Component {

  constructor() {
    super();
    this.isChoiceDeletable = this.isChoiceDeletable.bind(this);
  }

  isChoiceDeletable() {
    const wait = this.props.wait;
    if(wait) return false;
    const choices = this.props.choices;
    if(choices.length > 2) return true;
    return false;
  }

  render() {
    const {value, index, wait, updateChoice, removeChoice} = this.props;
    return (
      <div key={index}>
        <div style={styles.container}>
          <TextField value={value}
                     onChange={(e, newValue) => updateChoice(newValue, index)}
                     hintText={`Choice ${index+1}`}
                     fullWidth={true}
                     underlineShow={false}
                     disabled={wait}
          />
          <IconButton style={styles.icon} 
                      onClick={() => removeChoice(index)}
                      disabled={!this.isChoiceDeletable()}>
            <ContentClear />
          </IconButton>
        </div>
        <Divider />
      </div>
    );
  }
}

Choice.propTypes = {
  value: React.PropTypes.string.isRequired,
  index: React.PropTypes.number.isRequired, 
  wait: React.PropTypes.bool.isRequired,
  choices: React.PropTypes.array.isRequired,
  updateChoice: React.PropTypes.func.isRequired,
  removeChoice: React.PropTypes.func.isRequired
};

export default Choice;
