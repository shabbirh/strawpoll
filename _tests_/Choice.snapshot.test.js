import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import Choice from '../src/components/shared/Choice';
import renderer from 'react-test-renderer';
import React from 'react';

test('Choice component renders the choice correctly', () => {
  const rendered = renderer.create(
    <MuiThemeProvider>
      <Choice value={'test'}
              index={0}
              wait={false}
              choices={[]}
              updateChoice={()=>null}
              removeChoice={()=>null} />
    </MuiThemeProvider>
  );
  expect(rendered.toJSON()).toMatchSnapshot();
});