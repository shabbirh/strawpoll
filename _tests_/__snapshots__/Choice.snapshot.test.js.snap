exports[`test Choice component renders the choice correctly 1`] = `
<div>
  <div
    style={
      Object {
        "paddingRight": "48px",
        "position": "relative",
      }
    }>
    <div
      className={undefined}
      style={
        Object {
          "backgroundColor": "transparent",
          "cursor": "auto",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 16,
          "height": 48,
          "lineHeight": "24px",
          "muiPrepared": true,
          "position": "relative",
          "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "width": "100%",
        }
      }>
      <div
        style={
          Object {
            "bottom": 12,
            "color": "rgba(0, 0, 0, 0.3)",
            "muiPrepared": true,
            "opacity": 0,
            "position": "absolute",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          }
        }>
        Choice 1
      </div>
      <input
        disabled={false}
        id="TextField-0"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "appearance": "textfield",
            "backgroundColor": "rgba(0,0,0,0)",
            "border": "none",
            "color": "rgba(0, 0, 0, 0.87)",
            "cursor": "inherit",
            "font": "inherit",
            "height": "100%",
            "muiPrepared": true,
            "outline": "none",
            "padding": 0,
            "position": "relative",
            "width": "100%",
          }
        }
        type="text"
        value="test" />
    </div>
    <button
      disabled={true}
      href={undefined}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOut={[Function]}
      onTouchStart={[Function]}
      onTouchTap={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "background": "none",
          "border": 10,
          "boxSizing": "border-box",
          "cursor": "default",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 0,
          "fontWeight": "inherit",
          "height": 48,
          "margin": 0,
          "muiPrepared": true,
          "outline": "none",
          "overflow": "visible",
          "padding": 12,
          "position": "absolute",
          "right": 0,
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "verticalAlign": null,
          "width": 48,
          "zIndex": 1,
        }
      }
      tabIndex={-1}
      type="button">
      <svg
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.3)",
            "cursor": "not-allowed",
            "display": "inline-block",
            "fill": "rgba(0, 0, 0, 0.3)",
            "height": 24,
            "muiPrepared": true,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "width": 24,
          }
        }
        viewBox="0 0 24 24">
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z" />
      </svg>
    </button>
  </div>
  <hr
    style={
      Object {
        "backgroundColor": "#e0e0e0",
        "border": "none",
        "height": 1,
        "margin": 0,
        "marginLeft": 0,
        "marginTop": -1,
        "muiPrepared": true,
      }
    } />
</div>
`;
